sdks/python/apache_beam/typehints/row_type.py

```diff
@@ -160,3 +168,50 @@ def __repr__(self):
 
   def get_type_for(self, name):
     return dict(self._fields)[name]
+
+
+class GeneratedClassRowTypeConstraint(RowTypeConstraint):
+  """Specialization of RowTypeConstraint which relies on a generated user_type.
+
+  Since the generated user_type cannot be pickled, we supply a custom __reduce__
```

Comment:
> the generated user_type cannot be pickled

Can you please explain why this is the case?

Reply:
I actually don't fully understand it, but it's been a consistent issue with the Schema code. Each pickle library (built-in, dill, cloudpickle) fails for a different reason. I filed https://github.com/apache/beam/issues/22714 to track this, and added a (skipped) test.